src/RadixDlt.NetworkGateway.PostgresIntegration/Services/PendingTransactionPrunerService.cs

```diff
@@ -82,94 +83,63 @@ internal class PendingTransactionPrunerService : IPendingTransactionPrunerServic
 {
     private readonly IDbContextFactory<ReadWriteDbContext> _dbContextFactory;
     private readonly IOptionsMonitor<MempoolOptions> _mempoolOptionsMonitor;
-    private readonly ISystemStatusService _systemStatusService;
     private readonly ILogger<PendingTransactionPrunerService> _logger;
     private readonly IEnumerable<IPendingTransactionPrunerServiceObserver> _observers;
     private readonly IClock _clock;
 
     public PendingTransactionPrunerService(
         IDbContextFactory<ReadWriteDbContext> dbContextFactory,
         IOptionsMonitor<MempoolOptions> mempoolOptionsMonitor,
-        ISystemStatusService systemStatusService,
         ILogger<PendingTransactionPrunerService> logger,
         IEnumerable<IPendingTransactionPrunerServiceObserver> observers,
         IClock clock)
     {
         _dbContextFactory = dbContextFactory;
         _mempoolOptionsMonitor = mempoolOptionsMonitor;
-        _systemStatusService = systemStatusService;
         _logger = logger;
         _observers = observers;
         _clock = clock;
     }
 
     public async Task PrunePendingTransactions(CancellationToken token = default)
     {
+        var mempoolConfiguration = _mempoolOptionsMonitor.CurrentValue;
+
+        var pruneIfLastGatewaySubmissionBefore = _clock.UtcNow - mempoolConfiguration.PruneMissingTransactionsAfterTimeSinceLastGatewaySubmission;
+
+        Expression<Func<PendingTransaction, bool>> canPruneExpression = t =>
+            /* Prune if it was submitted a while ago */
+            t.LastSubmittedToGatewayTimestamp < pruneIfLastGatewaySubmissionBefore;
+
         await using var dbContext = await _dbContextFactory.CreateDbContextAsync(token);
 
-        var rawCountByStatus = await dbContext
+        var countByStatus = await dbContext
             .PendingTransactions
-            .GroupBy(t => t.Status)
-            .Select(g => new { Status = g.Key, Count = g.Count() })
+            .GroupBy(canPruneExpression)
+            .Select(g => new PendingTransactionCountByPruneStatus(g.Key, g.Count()))
             .ToListAsync(token);
```

Comment:
IIRC EF Core struggled a bit when we used custom type (`PendingTransactionCountByPruneStatus`) in select expression. Hence the `rawCountByStatus` followed by another LINQ expression. Could you confirm this refactored code does not result in warnings?

Reply:
Seems to be okay - can't see any logs... What warnings specifically were you getting?

Reply:
Honestly speaking don't remember what was the exact issue. It complaint about this custom type in `.Select` - this is why we used this two-step procedure to first fetch the data into anonymous object and then to map it into desired custom type. Maybe with recent EF Core upgrades they solve this inconvenience? If it works just fine I'm more than happy to have it coded in this simplified fashion! :)

Reply:
Seems to work fine :D